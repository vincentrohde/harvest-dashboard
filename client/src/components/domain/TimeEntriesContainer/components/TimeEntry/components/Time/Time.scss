.Time {
    font-size: 22px;
    color: #ff6f5e;
    margin: 10px 0 0;

    i {
        font-size: 0.9em;
    }
}
