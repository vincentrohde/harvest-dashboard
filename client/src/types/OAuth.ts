export type oAuthType = {
    access_token: string;
    account_id: string;
    expiration_date: string;
    refresh_token: string;
};
