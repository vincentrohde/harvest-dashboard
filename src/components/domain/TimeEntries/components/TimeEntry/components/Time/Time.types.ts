export interface TimeProps {
    hoursAndMinutes: string;
    isActive: boolean;
    toggleActive: () => void;
}
