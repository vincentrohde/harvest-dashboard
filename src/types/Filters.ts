export interface FiltersInterface {
    dateRange: string[] | 1;
}
