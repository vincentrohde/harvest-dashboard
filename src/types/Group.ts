export type groupsType = GroupInterface[];

export interface GroupInterface {
    id: number;
    name: string;
}