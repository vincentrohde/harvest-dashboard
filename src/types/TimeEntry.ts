import { ProjectInterface } from './Project';
import { TaskInterface } from './Task';

export type timeEntriesType = TimeEntryInterface[];

export interface ShallowTimeEntryInterface {
    hours: number | string;
    id: number;
    notes: string;
    spent_date: string;
    is_running: boolean;
    project: ProjectInterface;
    task: TaskInterface;
}

export interface TimeEntryInterface extends ShallowTimeEntryInterface {
    hours: number;
}

export interface ShallowSubmissionEntryInterface {
    notes: ShallowTimeEntryInterface['notes'];
    spent_date: ShallowTimeEntryInterface['spent_date'];
    hours: ShallowTimeEntryInterface['hours'];
    project_id: number | string;
    task_id: number | string;
}

export interface TimeEntrySubmissionInterface extends ShallowSubmissionEntryInterface {
    hours: TimeEntryInterface['hours'];
    project_id: number;
    task_id: number;
}