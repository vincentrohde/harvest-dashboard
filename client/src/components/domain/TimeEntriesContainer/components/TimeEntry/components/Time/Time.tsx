// Libs
import React from 'react';
import { Icon } from 'semantic-ui-react';

// Types
import { TimeProps } from './Time.types';

// Styles
import './Time.scss';

const Time = ({ hoursAndMinutes, isActive, toggleActive }: TimeProps) => (<h3 className="Time">
    <Icon className="js-toggle-active"
        // @ts-ignore
          name={`clock${isActive ? '' : ' outline'}`} onClick={toggleActive}/>
    { hoursAndMinutes }
</h3>);

export default Time;
