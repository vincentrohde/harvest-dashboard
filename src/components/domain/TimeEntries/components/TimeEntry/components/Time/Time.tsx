// Libs
import React from 'react';
import { Icon } from 'semantic-ui-react';

// Types
import { TimeProps } from './Time.types';

// Styles
import styles from './Time.module.scss';

const Time = ({ hoursAndMinutes, isActive, toggleActive }: TimeProps) => (<h3 className={styles.Time}>
    <Icon className="js-toggle-active"
        // @ts-ignore
          name={`clock${isActive ? '' : ' outline'}`} onClick={toggleActive}/>
    { hoursAndMinutes }
</h3>);

export default Time;
