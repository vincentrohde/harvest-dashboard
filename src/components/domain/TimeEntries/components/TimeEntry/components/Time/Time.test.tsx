// Libs
import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

// Components
import Time from './Time';

// Types
import { TimeProps } from './Time.types';

describe('<Time />', () => {
    it('renders inactive', () => {
        const mockData: TimeProps = {
            hoursAndMinutes: '1:00',
            isActive: false,
            toggleActive: () => null
        }

        const { asFragment } = render(
            <Time {...mockData} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders active', () => {
        const mockData: TimeProps = {
            hoursAndMinutes: '1:00',
            isActive: true,
            toggleActive: () => null
        }

        const { asFragment } = render(
            <Time {...mockData} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
})
