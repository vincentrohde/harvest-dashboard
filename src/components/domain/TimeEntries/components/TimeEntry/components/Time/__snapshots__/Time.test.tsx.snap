// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Time /> renders 1`] = `
<DocumentFragment>
  <h3
    class="Time"
  >
    <i
      aria-hidden="true"
      class="clock outline icon js-toggle-active"
    />
    1:00
  </h3>
</DocumentFragment>
`;

exports[`<Time /> renders active 1`] = `
<DocumentFragment>
  <h3
    class="Time"
  >
    <i
      aria-hidden="true"
      class="clock icon js-toggle-active"
    />
    1:00
  </h3>
</DocumentFragment>
`;

exports[`<Time /> renders inactive 1`] = `
<DocumentFragment>
  <h3
    class="Time"
  >
    <i
      aria-hidden="true"
      class="clock outline icon js-toggle-active"
    />
    1:00
  </h3>
</DocumentFragment>
`;
